cmd/machine-api-migration/main.go

```diff
@@ -153,6 +166,32 @@ func main() {
 		os.Exit(0)
 	}
 
+	k8sclient := mgr.GetClient()
+	infra := &configv1.Infrastructure{}
+
+	err = k8sclient.Get(stop, client.ObjectKey{Name: controllers.InfrastructureResourceName}, infra)
+	if err != nil {
+		klog.Errorf("failed to fetch infrastructure: %s", err)
+		os.Exit(1)
+	}
```

Comment:
We normally use inline if error handling if the func just returns an error, to minimise the err scope.